diff --git a/console/ratingprofile_set.go b/console/ratingprofile_set.go
index a86aae05..18c347ae 100644
--- a/console/ratingprofile_set.go
+++ b/console/ratingprofile_set.go
@@ -18,7 +18,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>
 
 package console
 
-import "github.com/cgrates/cgrates/utils"
+import "github.com/cgrates/cgrates/apier/v1"
 
 func init() {
 	c := &CmdSetRatingProfile{
@@ -33,7 +33,7 @@ func init() {
 type CmdSetRatingProfile struct {
 	name      string
 	rpcMethod string
-	rpcParams *utils.TPRatingProfile
+	rpcParams *v1.AttrSetRatingProfile
 	rpcResult string
 	*CommandExecuter
 }
@@ -48,7 +48,7 @@ func (self *CmdSetRatingProfile) RpcMethod() string {
 
 func (self *CmdSetRatingProfile) RpcParams(reset bool) interface{} {
 	if reset || self.rpcParams == nil {
-		self.rpcParams = &utils.TPRatingProfile{}
+		self.rpcParams = &v1.AttrSetRatingProfile{}
 	}
 	return self.rpcParams
 }
