diff --git a/engine/tp_reader.go b/engine/tp_reader.go
index 25ce0622..e8a73aca 100755
--- a/engine/tp_reader.go
+++ b/engine/tp_reader.go
@@ -2074,6 +2075,7 @@ func (tpr *TpReader) IsValid() bool {
 }
 
 func (tpr *TpReader) WriteToDatabase(flush, verbose, disable_reverse bool) (err error) {
+	disable_reverse = true
 	if tpr.dm.dataDB == nil {
 		return errors.New("no database connection")
 	}
@@ -2086,6 +2088,10 @@ func (tpr *TpReader) WriteToDatabase(flush, verbose, disable_reverse bool) (err
 		log.Println("Destinations:", len(tpr.destinations))
 	}
 	for _, d := range tpr.destinations {
+		if destinationExists, err := tpr.dm.HasData(utils.DESTINATION_PREFIX, d.Id); err == nil && destinationExists {
+			continue
+		}
+		disable_reverse = false
 		err = tpr.dm.DataDB().SetDestination(d, utils.NonTransactional)
 		if err != nil {
 			return err
