diff --git a/engine/cdrs.go b/engine/cdrs.go
index 46d8a047..7feab366 100644
--- a/engine/cdrs.go
+++ b/engine/cdrs.go
@@ -402,11 +402,15 @@ func (self *CdrServer) rateCDR(cdr *CDR) ([]*CDR, error) {
 	if cdr.RequestType == utils.META_NONE {
 		return nil, nil
 	}
+
+	// search in sm_cost only for first rating of prepaid calls
+	isFirstRating := cdr.Cost == -1 && cdr.RunID == "*default"
+
 	cdr.ExtraInfo = "" // Clean previous ExtraInfo, useful when re-rating
 	var cdrsRated []*CDR
 	_, hasLastUsed := cdr.ExtraFields[utils.LastUsed]
 	if utils.IsSliceMember([]string{utils.META_PREPAID, utils.PREPAID}, cdr.RequestType) &&
-		(cdr.Usage != 0 || hasLastUsed) { // ToDo: Get rid of PREPAID as soon as we don't want to support it backwards
+		(cdr.Usage != 0 || hasLastUsed) && isFirstRating { // ToDo: Get rid of PREPAID as soon as we don't want to support it backwards
 		// Should be previously calculated and stored in DB
 		fib := utils.Fib()
 		var smCosts []*SMCost
